app/controllers/one_login_controller.rb

```diff
@@ -0,0 +1,89 @@
+class OneLoginController < ApplicationController
+  before_action :one_login_enabled
+  before_action :set_sentry_context, only: :callback
+
+  def callback
+    auth = request.env['omniauth.auth']
+    session[:onelogin_id_token] = auth&.credentials&.id_token
+    candidate = OneLoginUser.authentificate(auth)
+
+    sign_in(candidate, scope: :candidate)
+    candidate.update!(last_signed_in_at: Time.zone.now)
+
+    redirect_to candidate_interface_interstitial_path
+  rescue OneLoginUser::Error => e
+    session[:one_login_error] = e.message
+    redirect_to auth_onelogin_sign_out_path
+  end
+
+  def bypass_callback
+    one_login_user_bypass = OneLoginUserBypass.new(
+      token: request.env['omniauth.auth']&.uid,
+    )
+    candidate = one_login_user_bypass.authentificate
+
+    if one_login_user_bypass.valid? && candidate.present?
+      sign_in(candidate, scope: :candidate)
+      candidate.update!(last_signed_in_at: Time.zone.now)
+
+      redirect_to candidate_interface_interstitial_path
+    else
+      flash[:warning] = one_login_user_bypass.errors.full_messages.join('\n')
+      redirect_to candidate_interface_create_account_or_sign_in_path
+    end
+  end
+
+  def sign_out
+    id_token = session[:onelogin_id_token]
+    one_login_error = session[:one_login_error]
+    reset_session
+
+    session[:one_login_error] = one_login_error
```

Comment:
We need to persist this one_login_error value in the session throughout the redirects and after the `reset_session`.

The `one_login_error` is used to redirect the user to the 500 page and log errors in sentry.

I choose to reset the session here and not in sign_out_complete just in case one_login fails to come back to us, so we logout the user on our end at least in case one_login doesn't come back to us in sign_out_complete